lms/static/js/spec/edxnotes/views/note_item_spec.js

```diff
@@ -72,6 +72,21 @@ define([
             expect(view.$('span.reference-tags')[1]).toContainText('Second');
         });
 
+        it('should highlight tags & text if they have elasticsearch formatter', function() {
+            var view = getView({
+                tags: ["First", "{elasticsearch_highlight_start}Second{elasticsearch_highlight_end}"]
```

Comment:
Can you add an example to demonstrate that any HTML in the text does get appropriately escaped. It would also be worth having some Unicode characters and verify that they come through correctly.